### ISSUE TITLE
Doesn't work with GenericServices 3.1.0

### ISSUE THREAD

**Issue by marius-stanescu** [JonPSmith/EfCore.GenericServices.AspNetCore#3]
I updated the GenericServices and GenericBizRunner packages to 3.1.0 and this library doesn't work anymore. I get the following error:

> Reference to type 'IStatusGeneric' claims it is defined in 'GenericServices', but it could not be found

**Comment by JonPSmith** [comment]
Hi @marius-stanescu,

Yes, there is a warning about that in the NuGet notes (which no one reads :)). You need to add the package `GenericServices.StatusGeneric` to your business logic, and any other projects that complain.

**Comment by JonPSmith** [comment]
PS. All my libs GenericServices, GenericBizRunner and GenericEventRunner now use the same status type. This allows you to combine errors from one to another. This is especially useful when using  GenericServices or GenericBizRunner with GenericEventRunner underneath. 

I need to write an article on that, but I'm pretty busy.

**Comment by marius-stanescu** [comment]
Hi @JonPSmith 

I read the warning in the NuGet notes :) and I installed the `GenericServices.StatusGeneric` package, but it still doesn't work. The errors are related to the `Response` method in the `EfCore.GenericServices.AspNetCore` package. That's why I created an issue here. I think you have to update this package as well.

**Comment by rfcdejong** [comment]
> Hi @JonPSmith
> 
> I read the warning in the NuGet notes :) and I installed the `GenericServices.StatusGeneric` package, but it still doesn't work. The errors are related to the `Response` method in the `EfCore.GenericServices.AspNetCore` package. That's why I created an issue here. I think you have to update this package as well.

I'm using .NET Core 3.1 with the following packagereferences, what are you using? This works fne

    <PackageReference Include="EfCore.GenericBizRunner" Version="4.0.0" />
    <PackageReference Include="EfCore.GenericEventRunner" Version="1.1.0" />
    <PackageReference Include="EfCore.GenericServices.AspNetCore" Version="4.0.0" />

**Comment by marius-stanescu** [comment]
I'm using .NET Core 2.2
With these versions it works
```
<PackageReference Include="EfCore.GenericBizRunner" Version="4.0.0" />
<PackageReference Include="EfCore.GenericServices" Version="3.0.0" />
<PackageReference Include="EfCore.GenericServices.AspNetCore" Version="4.0.0" />
```
But when I upgrade to the following, I start getting the error
```
<PackageReference Include="EfCore.GenericBizRunner" Version="4.1.0" />
<PackageReference Include="EfCore.GenericServices" Version="3.1.0" />
<PackageReference Include="EfCore.GenericServices.AspNetCore" Version="4.0.0" />
```

**Comment by rfcdejong** [comment]
Do you need really EfCore.GenericServices ? As I don't have it at all.

**Comment by marius-stanescu** [comment]
Yes, I need both GenericServices and GenericBizRunner. I think the problem is that these 2 packages were updated to use a different GenericStatus class, but GenericServices.AspNetCore was not updated.

**Comment by rfcdejong** [comment]
> Yes, I need both GenericServices and GenericBizRunner. I think the problem is that these 2 packages were updated to use a different GenericStatus class, but GenericServices.AspNetCore was not updated.

Seems EfCore.GenericServices.AspNetCore it is hard referenced to 3.0.0

https://github.com/JonPSmith/EfCore.GenericServices.AspNetCore/blob/master/GenericServices.AspNetCore/GenericServices.AspNetCore.csproj#L9

Maybe it should be changed into using wildcards, for example.. but that is up to JonPSmith

`<PackageReference Include="EfCore.GenericServices" Version="3.*" />`

**Comment by JonPSmith** [comment]
OK,

I haven't updated GenericServices.AspNetCore because @rfcdejong has raised [an issue](https://github.com/JonPSmith/EfCore.GenericServices.AspNetCore/issues/2) with ASP.NET Core, which I haven't the time to look at the moment. This work is done in my spare time.

**The solution is to drop back to GenericServices version 3.0.0 and GenericBizRunner version 4.0.0.** The only change in GenericBizRunner 4.1.0 is swapping to the GenericServices.StatusGeneric. GenericServices 3.1.0 swaps to GenericServices.StatusGeneric and adds one new feature, but no changes any existing features.

**Comment by JonPSmith** [comment]
PS. I have added a warning about this in the README file in GenericServices  and GenericBizRunner .

**Comment by marius-stanescu** [comment]
> GenericServices 3.1.0 swaps to GenericServices.StatusGeneric and adds one new feature, but no changes any existing features.

Yes, I need that feature that was added in GenericServices 3.1.0, for saving nested Dtos. That's why I tried to upgrade to 3.1.0.

**Comment by JonPSmith** [comment]
The other alternative is to copy out the code you use in GenericServices.AspNetCore into your own code. There isn't a lot of code in GenericServices.AspNetCore so should be easy.

That's the good thing about open-source - its there for you to copy.

**Comment by JonPSmith** [comment]
Ho @marius-stanescu,

I have finally got around to creating a preview  [EfCore.AspNetCore 4.0.0-preview001](https://www.nuget.org/packages/EfCore.GenericServices.AspNetCore/4.0.0-preview001) NuGet package. This ONLY works with NET 3.0 and above.

If you can try it then please do and let me know if it works. If you have another solution or don't have time then let me know. 

Sorry for the delay - I updated the whole Repo to NET Core 3.1 to check out a problem and forgot that I had updated the code on another laptop. The Merge was horrible, so I had to rebuild it all again.

**Comment by marius-stanescu** [comment]
Hey @JonPSmith ,

Sorry, I don't think I'll have the time to try it. I gave up on using this package. I wrote my own `Response` extension methods.

My project is on .NET Core 2.2 anyway, so it probably doesn't work anyway.

**Comment by JonPSmith** [comment]
Hi @marius-stanescu,

I fully understand and you are right - it doesn't (currently) support 2.2. I thought everyone had moved to NET 3, but I was obviously wrong. I need to think about that.

**Comment by JonPSmith** [comment]
I have released version 4.1.0 which works with my GenericServices and GenericBizRunner libraries. It supports both EF Core 2.1 and EF Core 3.0.

Sorry for the long delay to getting this done.
